src/celeritas/em/msc/UrbanMscScatter.hh

```diff
@@ -459,28 +514,36 @@ CELER_FUNCTION real_type UrbanMscScatter::simple_scattering(
  * \param true_path the true step length.
  */
 CELER_FUNCTION
-real_type UrbanMscScatter::compute_theta0(real_type true_path) const
+real_type UrbanMscScatter::compute_theta0() const
 {
-    real_type invbetacp
-        = std::sqrt((inc_energy_ + mass_) * (end_energy_ + mass_)
-                    / (inc_energy_ * (inc_energy_ + 2 * mass_) * end_energy_
-                       * (end_energy_ + 2 * mass_)));
+    real_type true_path = max(limit_min_, true_path_);
     real_type y = true_path / rad_length_;
 
     // Correction for the positron
     if (is_positron_)
     {
-        real_type tau = std::sqrt(inc_energy_ * end_energy_) / mass_;
-        y *= this->calc_positron_correction(tau);
+        y *= this->calc_positron_correction(
+            std::sqrt(inc_energy_ * end_energy_) / mass_);
     }
 
     // TODO for hadrons: multiply abs(charge)
+    real_type invbetacp
+        = std::sqrt((inc_energy_ + mass_) * (end_energy_ + mass_)
+                    / (inc_energy_ * (inc_energy_ + 2 * mass_) * end_energy_
+                       * (end_energy_ + 2 * mass_)));
     real_type theta0 = value_as<Energy>(c_highland()) * std::sqrt(y)
                        * invbetacp;
 
     // Correction factor from e- scattering data
     theta0 *= (msc_.coeffth1 + msc_.coeffth2 * std::log(y));
 
+    if (true_path_ < limit_min_)
+    {
+        // Correct for non-MSC-limited path lengths
+        theta0 *= std::sqrt(true_path_ / limit_min_);
+    }
+
+    CELER_ENSURE(theta0 >= 0);
```

Comment:
This seems a hard condition as theta0 can be a very small negative value (for a very large radiation length, ~1e+17, galactic material?).

Reply:
In that case, wouldn't the "skip_sampling" condition `true_path_ <= lambda_ * params_.tau_small` prevent us from even calculating theta0?

Reply:
It may be true (i.e, should be self-consistent), but the negative case is due to the material (radiation length) dependent correction to the apparently non-detector (galactic or gaseous?) volume of which radiation length is humongous (probably not physically meaningful for the msc scattering, but nothing practically wrong as it is close to 0-) while the skip_sampling condition is not always hold with the corresponding lambda (even though the m.p.f value is still big, but smaller than true_path * tau_small).  Anyway, as the theta0 itself is based on the 2nd momentum of the angular differential cross section, I do agree that it should be always positive, but should not impose the requirement due to the rather ah-hoc empirical correction (and the evaluated lambda for this non-detector material, whatever it is).

Reply:
Given this empirical model of theta0 that's implemented, the result should be strictly positive and it improves code clarity (and may catch a bug) if we mark it as such.